satellite/rewards/partners.go

```diff
@@ -65,3 +66,30 @@ func LoadPartnerInfos() Partners {
 		},
 	}
 }
+
+// GeneratePartnerLink returns base64 encoded partner referral link
+func GeneratePartnerLink(offerName string) ([]string, error) {
+	referralInfo := &referralInfo{UserID: "", PartnerID: GetPartnerIDByName(offerName)}
+	refJSON, err := json.Marshal(referralInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	domians := getTardigradeDomains()
+	referralLinks := make([]string, len(domians))
+
+	for i, url := range domians {
+		referralLinks[i] = url + base64.StdEncoding.EncodeToString(refJSON)
```

Comment:
this makes an assumption that the url has a `/` at the end, which is not always safe to do. I would recommend using `path.Join` from this package: https://golang.org/pkg/path/

Reply:
Also, you can do `base64.StdEncoding.EncodeToString(refJSON)` outside of the loop, so it does not need to be evaluated separately for each iteration